src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java

```diff
@@ -351,6 +352,17 @@ private <T, C extends GitSCMSourceContext<C, R>, R extends GitSCMSourceRequest>
         return doRetrieve(retriever, context, listener, prune, getOwner(), delayFetch);
     }
 
+    /**
+     * Returns false if a non-TLS protocol is used when FIPS mode is enabled.
+     * @param credentialsId any credentials (can be {@code null})
+     * @param remoteUrl the git remote url
+     * @return {@code false} if using any credentials with a non TLS protocol with FIPS mode activated
+     * @see FIPS140#useCompliantAlgorithms()
+     */
+    public static boolean isFIPSCompliantTLS(String credentialsId, String remoteUrl) {
+        return !FIPS140.useCompliantAlgorithms() || !StringUtils.isNotEmpty(credentialsId) || (!StringUtils.startsWith(remoteUrl, "http:") && !StringUtils.startsWith(remoteUrl, "git:"));
```

Comment:
I realize that this is a move rather from one file to another, but I detected a double negative while reading this a  second time.  I find double negatives hard to read.  Optional, since the code as implemented is correct and my preference to not use a double negative is just a personal preference.  Apologies that I didn't detect this on the first code review.
```suggestion
        return !FIPS140.useCompliantAlgorithms() || StringUtils.isEmpty(credentialsId) || (!StringUtils.startsWith(remoteUrl, "http:") && !StringUtils.startsWith(remoteUrl, "git:"));
```